cpp/src/lists/drop_list_duplicates.cu

```diff
@@ -426,132 +400,133 @@ struct table_row_comparator_fn {
- *         contains their corresponding list offsets.
+ * @brief Extract list entries and their corresponding (1-based) list indices ignoring duplicate
+ * entries.
  */
-std::vector<std::unique_ptr<column>> get_unique_entries_and_list_offsets(
-  column_view const& all_lists_entries,
-  column_view const& entries_list_offsets,
+std::vector<std::unique_ptr<column>> get_unique_entries_and_list_indices(
+  column_view const& keys_entries,
+  std::optional<column_view> const& values_entries,
+  rmm::device_uvector<size_type> const& entries_list_indices,
   null_equality nulls_equal,
   nan_equality nans_equal,
+  duplicate_keep_option keep_option,
   rmm::cuda_stream_view stream,
   rmm::mr::device_memory_resource* mr)
 {
-  auto const num_entries = all_lists_entries.size();
+  auto const num_entries = keys_entries.size();
 
   // Allocate memory to store the indices of the unique entries.
-  auto unique_indices     = rmm::device_uvector<offset_type>(num_entries, stream);
+  auto unique_indices     = rmm::device_uvector<size_type>(num_entries, stream);
   auto const output_begin = unique_indices.begin();
-  auto const output_end   = type_dispatcher(all_lists_entries.type(),
-                                          get_unique_entries_dispatch{},
-                                          entries_list_offsets.begin<offset_type>(),
-                                          all_lists_entries,
+  auto const output_end   = type_dispatcher(keys_entries.type(),
+                                          get_indices_of_unique_entries_dispatch{},
+                                          entries_list_indices.begin(),
+                                          keys_entries,
                                           num_entries,
                                           output_begin,
                                           nulls_equal,
                                           nans_equal,
-                                          all_lists_entries.has_nulls(),
+                                          keys_entries.has_nulls(),
+                                          keep_option,
                                           stream);
 
-  auto gather_map = column_view(data_type{type_to_id<offset_type>()},
-                                static_cast<size_type>(thrust::distance(output_begin, output_end)),
-                                unique_indices.data());
-
-  // Collect unique entries and entry list offsets.
-  // The new null_count and bitmask of the unique entries will also be generated
-  // by the gather function.
-  return cudf::detail::gather(table_view{{all_lists_entries, entries_list_offsets}},
+  // The indices of the unique key entries will be used as a gather map to collect keys and values
+  // entries.
+  auto const gather_map =
+    column_view(data_type{type_to_id<size_type>()},
+                static_cast<size_type>(thrust::distance(output_begin, output_end)),
+                unique_indices.data());
```

Comment:
There's a `column_view(device_uvector)` ctor that makes this cleaner.

Reply:
Since that ctor does not exist yet, I filed an issue for it to be added later: https://github.com/rapidsai/cudf/issues/9656

Reply:
There is a `cudf::detail::gather` API that takes a device-span so you would not necessarily need to create a `column_view` wrapper here.